src/CosmoShips.sol

```diff
@@ -35,19 +44,43 @@ contract CosmoShips is ERC721, AccessControl, ReentrancyGuard, AttributeEncoder,
         mintPrice = _newPrice;
     }
 
-    function maintenance(uint256 _tokenId, uint256 _newAttribute) external onlyRole(MAINTENANCE_ROLE) {
-        attributes[_tokenId] = _newAttribute;
+    function maintenance(uint256 tokenId, uint256 _newAttribute) external onlyRole(MAINTENANCE_ROLE) {
+        attributes[tokenId] = _newAttribute;
+    }
+
+    function mint(uint256 _attributes, bytes32[] calldata _proof) external payable nonReentrant {
+        require(msg.value == mintPrice, "Incorrect payment sent");
+        require(!tokenMinted[_shipIdToMint], "Token already minted");
```

Comment:
Now that we have serial minting, this can be removed entirely. 
IDs minted are IDs <_nextTokenIdToMint